docs/getting-started/auth-self-defence-checks.adoc

```diff
@@ -0,0 +1,57 @@
+= Self Defence Checks
+
+As part of the Auth SDK module, there is a self defence check feature.  This feature offers multiple checks that can be used to analyse the secuirty of the device.
+
+== Prequisite's
+
+. Import and initialise the Auth SDK module - link:auth.adoc[AeroGear Services Auth SDK].
+
+== Usage
+
+To begin using the self defence checks in your app, you'll need to retrieve an instance of `SecurityService`. The `SecurityService` can be retrieved using the `MobileCore#getInstance` method.
+
+[source, java]
+----
+SecurityService securityService = MobileCore.getInstance(SecurityService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `SecurityService`.
+
+=== Invoking a Single Self Defence Check
+The `SecurityService#check` method can be used to run a single secuirty check.  It expects a `SecurityCheckType` paramater and returns `SecurityCheckResult`.
+
+[source, java]
+----
+SecurityCheckResult result = securityService.check(SecurityCheckType.<check_type>);
+----
+
+See all available `SecurityCheckType` link:auth-self-defence-javadoc.adoc[here].
+
+=== Invoking Multiple Self Defence Checks
+To run multiple self defence checks you'll first need to invoke the `SecurityService#getCheckExecutor` method.  This methods returns `SecurityCheckExecutor` where checks can be added (chained) to the executor.
+
+[source, java]
+----
+SecurityCheckExecutor checkExecutor = securityService.getCheckExecutor();
+----
+
+The `SecurityCheckExecutor#addCheck` method can be used to add multiple self defence checks to the `checkExecutor`.  The `addCheck` function accepts `SecurityCheckType` as a parameter and returns the `SecurityCheckExecutor`. 
+
+[source, java]
+----
+// add two checks to the checkExecutor
+checkExecutor.addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>);
+----
+
+To run all the security self defence checks that have been added to the `checkExecutor` the `SecurityCheckExecutor#execute` method must be invoked.  This method returns an array of `SecurityCheckResult`.
+
+[source, java]
+----
+SecurityCheckResult results[] = checkExecutor.execute();
+----
+
+The `SecurityCheckExecutor` allows for multiple checks to be chained so the above can be refactored to:
+[source, java]
+----
+SecurityCheckResult results[] = securityService.getCheckExecutor().addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>).execute();
```

Comment:
Why do we return an `array` - instead of a collection ? Is that something that Android usually does? 
IMO using arrays kinda makes the code read like written in 1998. 

Anyone knows? @wei-lee @danielpassos

Reply:
@matzew  I have no idea! Nothing Android specific

Reply:
@matzew @danielpassos That has been addressed here [#90](https://github.com/aerogear/aerogear-android-sdk/pull/90)

Reply:
@rachael-oregan @matzew See my previous comment